### PR TITLE
feat: verify chain parameter for `Subgraph` and dump subgraph urls to json

### LINKED ISSUE #239: get_subgraph_url is not always working and errors are hard to understand.
See this action run for an example. 

https://github.com/BalancerMaxis/protocol_fee_allocator/actions/runs/8394398285/job/22991436942
https://github.com/BalancerMaxis/protocol_fee_allocator/pull/149/commits/1e960c0166e953c637028a5237c97b84ae9c9348
I think the problem was actually that I needed to send chain.value and not just chain.  Still working ot debug. 

takes a long time to figure out what was wrong.  
A few comments:

1: Not sure this should ever revert like this, or if so should be a lot more descriptive in it's error (I can't find the core subgraph for chain name "blah blah" when trying to query the frontend file.  You are passing in a subgraph name and a chain.  An error should come back stating which you one you passed in that was wrong, what you passed in so you can see why it is wrong, and maybe also where you can look to see what the valid inputs are.

2: It would be really nice to have a static/human reasdable json file of endpoints that you could look at to understand what availablep options were.  Could we use the logic to generate endpoints from various frontend files to build a json file and use the json for actual lookups like we do in other use cases?  This would make it a lot easier to understand how to work with the library.

### PR DESCRIPTION
**Pull request by gosuto-inzasheru**
closes #239 

- adds data verification on `chain` parameter when initiating `Subgraph`
- adds script that can be run manually to dump all subgraph urls to json

### CODE REVIEW

File: bal_addresses/subgraph.py [hunk 1]
```diff
@@ -31,7 +37,7 @@ def get_subgraph_url(self, subgraph="core") -> str:
             if chain == "zkevm":
                 return "https://subgraph.satsuma-prod.com/ab0804deff79/1xhub-ltd/aura-finance-zkevm/api"
             elif chain in ["avalanche"]:  # list of chains without an aura subgraph
-                return None
+                return ""
```
Comment (Tritium-VLK) [inline bal_addresses/subgraph.py:40]: Why this change?  Should think about a consistent no result in the library if possible.

Comment (Tritium-VLK) [reply bal_addresses/subgraph.py:40]: can we still return none and catch that in the gen_subgraph script?

File: gen_subgraph_urls.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    urls = {}
+
+    with open("extras/chains.json", "r") as f:
+        chains = json.load(f)
+    for chain in chains["CHAIN_IDS_BY_NAME"]:
+        if chain not in urls:
+            urls[chain] = {}
+        for subgraph_type in ["core", "gauges", "blocks", "aura"]:
+            subgraph = Subgraph(chain)
+            url = subgraph.get_subgraph_url(subgraph_type)
```
Comment (Tritium-VLK) [inline gen_subgraph_urls.py:15]: Could we add some logic that tests that all the URLs are basically working, at least don't return a 404 when running some kind of a simple/nothing query?

If they don't maybe somehow signal that in the outputs?

Comment (gosuto-inzasheru) [reply gen_subgraph_urls.py:18]: so this is one possible approach: 25b4cc3

Comment (Tritium-VLK) [reply gen_subgraph_urls.py:18]: Yeah looks good.  As you can see the aura subgra[hs are incomplete, but that may be due to the lack of complete subgraph coverage by aura or another schema or something.  

In lieu of any docs/clear record of where to find consistent subgraph endpoints per chain, maybe we should just not include the ecosystem stuff in our library?   Will open another ticket to discuss this.
